src/libaktualizr/uptane/fetcher.cc

```diff
@@ -49,32 +49,54 @@ static size_t DownloadHandler(char* contents, size_t size, size_t nmemb, void* u
     auto event = std::make_shared<event::DownloadProgressReport>(ds->target, "Downloading", calculated);
     (*(ds->events_channel))(event);
   }
+  if (ds->fetcher->isPaused()) {
+    return written_size + 1;
+  }
   return written_size;
 }
 
+void Fetcher::setPause(bool pause) {
+  if (pause) {
+    pause_mutex_.lock();
+  } else {
+    pause_mutex_.unlock();
```

Comment:
Unlocking an unlocked mutex can throw.

Reply:
Good catch. Also we will get deadlock if call setPause(true) twice.

Reply:
Updated. Now it is impossible